### PR TITLE
Add LevelDB support to store upload state between program runs

### PR DESCRIPTION
**Pull request by maran**
Required this for some prototyping, figured you might want it in your own fork as well. It's quick and dirty; if you have any comments let me know.

### CODE REVIEW

**Comment by maran** [comment]
Forced pushed, should be in the green now.

File: README.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ Store is used to map an upload's fingerprint with the corresponding upload URL.
 | Name | Backend | Dependencies |
 |:----:|:-------:|:------------:|
 | MemoryStore | In-Memory | None |
+| LeveldbStore | LevelDB | None |
```
Comment (alexandrevicenzi) [inline README.md]: LevelDB is not std lib, so it has dependencies.

File: leveldbstore/store.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	"github.com/syndtr/goleveldb/leveldb"
+	"path/filepath"
+)
+
+type LeveldbStore struct {
+	db *leveldb.DB
+}
+
+func NewLeveldbStore() (tus.Store, error) {
+	home, err := homedir.Dir()
```
Comment (alexandrevicenzi) [inline leveldbstore/store.go]: Why not pass the path as an argument? What if I want another location, plus, it removes a random dependency. It also could be replaced by "os/user" package.

Comment (maran) [inline leveldbstore/store.go]: Just personal preference. There was no set precedence in the codebase yet on how to handle this. If you prefer a path I can change that.

Comment (alexandrevicenzi) [inline leveldbstore/store.go]: It's better, maybe someone one day could place in `/var` folder for example, or on Windows `C:\ProgramData`.

File: README.md (hunk 1 above)
Comment (maran) [inline README.md]: Depends on what you mean with "Dependencies", it won't require an external database such as Redis etc would. This is what I thought you meant with Dependencies.

Comment (alexandrevicenzi) [inline README.md]: I mean, when you go get it will get extra repositories. But it could also include other dependencies like Redis DB.

Comment (maran) [inline README.md]: How would you like me to document the dependencies? By supplying the import paths?

Comment (alexandrevicenzi) [inline README.md]: It could be `[goleveldb](https://github.com/syndtr/goleveldb)`.

**Comment by maran** [comment]
PR updated with requested changes.

**Pull request merged**